artemis-cli/src/test/java/org/apache/activemq/artemis/cli/commands/CreateTest.java

```diff
@@ -0,0 +1,119 @@
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.cli.commands;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+
+import org.apache.activemq.artemis.utils.XmlProvider;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+@RunWith(Parameterized.class)
+public class CreateTest {
+
+   private final String testName;
+   private String httpHost;
+   private boolean relaxJolokia;
+
+   public CreateTest(String testName, String httpHost, boolean relaxJolokia) {
+      this.testName = testName;
+      this.httpHost = httpHost;
+      this.relaxJolokia = relaxJolokia;
+   }
+
+   @Parameters
+   public static Collection<Object[]> testData() {
+      return Arrays.asList(new Object[][]{
+         {"Happy path + relaxJolokia", "sampledomain.com", true},
+         {"Happy path - relaxJolokia", "sampledomain.net", false},
+         {"Domain with dash + relaxJolokia", "sample-domain.co", true},
+         {"Domain with dash - relaxJolokia", "sample-domain.co.uk", false},
+         {"Domain with double dashes + relaxJolokia", "sample--domain.name", true},
+         {"Domain with double dashes - relaxJolokia", "sample--domain.biz", false},
+         {"Domain with leading dashes + relaxJolokia", "--sampledomain.company", true},
+         {"Domain with leading dashes - relaxJolokia", "--sampledomain.email", false},
+         {"Domain with trailing dashes + relaxJolokia", "sampledomain--.shop", true},
+         {"Domain with trailing dashes - relaxJolokia", "sampledomain--.java", false},
+      });
+   }
+
+   @Test
+   public void testWriteJolokiaAccessXmlCreatesValidXml() throws Exception {
+      Create c = new Create();
+      String source = Create.ETC_JOLOKIA_ACCESS_XML;
+      HashMap<String, String> filters = new LinkedHashMap<>();
+
+      filters.put("${http.host}", this.httpHost);
+
+      // This is duplicated from Create.java, but it's embedded into the middle of the class.
+      // TODO: Refactor that code to it's own method.
+      if (this.relaxJolokia) {
+         filters.put("${jolokia.options}", "<!-- option relax-jolokia used, so strict-checking will be removed here -->");
+      } else {
+         filters.put("${jolokia.options}", "<!-- Check for the proper origin on the server side, too -->\n" + "        <strict-checking/>");
+      }
+
+      Path temp = Files.createTempFile("jolokia-access-", ".xml");
+      try {
+         c.write("etc/" + source, temp.toFile(), filters, false, true);
```

Comment:
The test could avoid to duplicate embedded Create class stuff exposing setNoAutoTune, setHttpHost and setRelaxJolokia:
```suggestion
      TestActionContext context = new TestActionContext();
      File testInstance = new File(temporaryFolder.getRoot(), "test-instance");

      Create c = new Create();
      c.setNoAutoTune(true);
      c.setInstance(testInstance);
      c.setHttpHost(httpHost);
      c.setRelaxJolokia(relaxJolokia);
      c.execute(context);
```